### PR TITLE
Javascript Capstone Project

### PR DESCRIPTION
**Pull request by Samuelwanza**
@ctokoli  and @Samuelwanza  have conducted the following in our show display and comment javascript capstone project:
Use JavaScript to make websites dynamic and build basic single page apps.
Use ES6 syntax.
Use ES6 modules.
Use callbacks and promises.
Use webpack.
Apply JavaScript best practices and language style guides in code.
Use AAA pattern for unit tests.
Write units tests for a JavaScript app.
Follow Gitflow.
Solve simple git conflicts.
Send and receive data from an API.
Use API documentation.
Understand and use JSON.
Make JavaScript code asynchronous.
Perform a code review for a team member.

### CODE REVIEW

File: modules/appData.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const appId = 'PVsyyzJcjAI3kwRWNz7v';
```
Comment (peterrobert) [inline modules/appData.js:1]: - [ ] It will be great to ensure that you don't expose the AppID. This is sensitive information about the API and it shouldn't be committed to GitHub. This ensures security in your application. Just provide instructions on how to obtain the keys needed to use the API. Or use `env` to store your appID.

File: showCount.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+it('should count the number of shows', async () => {
+  const likes = await showLikes();
+  const count = likes.length;
+  expect(count).toEqual(15);
+});
+it('should count the number of comments for the show, Person of interest', async () => {
+  const comments = await getComments(2);
+  const count = comments.length;
+  expect(count).toEqual(9);
+});
```
Comment (peterrobert) [inline showCount.test.js:13]: - [ ] It will be great to ensure that the test cases are not failing. It seems your tests are failing kindly check below ⬇️ 
javascriptTest

- [ ] Kindly also Make sure that you have tests for all edge cases. Example: What if there are 0 items on the page (counter function should return 0)

**Review by peterrobert — CHANGES_REQUESTED** [review]
Hi @ctokoli  and @Samuelwanza , 

Very nice project. I love the way you have separated your code into modules and assigned specific tasks to specific modules 👏🏿 . However, there are a few changes to be made kindly check below ⬇️

### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work with your code structure is well divided and its easy to follow✔️
[ ] Great work on the design. It looks awesome. ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/appData.js (hunk 1 above)
Comment (wayungi) [reply modules/appData.js:1]: - [ ] As previously requested by the reviewer, Please do not 
commit confidential information about the API to GitHub. Provide instructions on how to obtain the keys needed to use the API or use the .env to store confidential data 👍🏾

File: showCount.test.js
Comment (wayungi) [inline showCount.test.js:1]: - [ ] Please note that you still have failing tests. The previous reviewer suggested that you correct. Please i suugest you do the same. 

test1
test2

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @Samuelwanza , 

image

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- [ ] All linter checks passing ✔️ 
- [ ] Descriptive PR body 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- Well-organized and clean project ✔️
- Professional readme file ✔️
- Good commit history ✔️

approv


# Status: Approved ✔️✔️✔️

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```
